### PR TITLE
STRY0017377 - Testing GAS call with Many Thresholds

### PR DESCRIPTION
**Pull request by emarinier**
**Story**: STRY0017377 - Write extensive unit/integration tests for GAS

**Description**: As a developer, I wish to have an extensive test suite (unit/integration tests) for GAS in order to verify this core component of deliverables for Enterics is working properly.

**Acceptance criteria**:

- [x] Expand tests for "gas call" (assign.py) to test for edge cases
  - [x] Test requirement that numbers on right side of address can never be smaller than left

### CODE REVIEW

File: tests/test_call.py [hunk 1]
```diff
@@ -347,6 +347,101 @@ def test_no_thresholds(tmp_path):
...
+    assert path.isdir(output_path)
+
+    # Clusters
+    clusters_path = path.join(output_path, "results.text")
+    assert path.isfile(clusters_path)
+    with open(clusters_path) as clusters_file:
+        clusters = csv.reader(clusters_file, delimiter="\t")
+
+        # The new f is the same as the existing a (1.1.1.1.1.1.1.1.1.1.1.1.1.1)
+        # The new g is the same as the existing e (1.1.1.3.3.4.4.4.4.4.4.4.4.4)
```
Comment (apetkau) [inline tests/test_call.py:385]: Could you add another test case where there are new cluster addresses created instead of assigning samples to existing cluster addresses? That way we can make sure that when new cluster addresses are created, the property of numbers to the right never being lower than numbers to the left holds true.

Comment (emarinier) [reply tests/test_call.py:385]: This was added after your comment, but this test from #42 adds a new address:

https://github.com/phac-nml/genomic_address_service/blob/1d390ea1395961ddeee5b79bdeb3be8178f243de/tests/test_call.py#L1679-L1755

Comment (emarinier) [reply tests/test_call.py]: I've changed the distances so that `f` remains the same (same as `a`), but `g` now is completely different: 488a7b9

```
query_id	ref_id	dist
f	a	0
f	b	17
f	c	21
f	d	31
f	e	23
f	f	0
f	g	23
g	a	123
g	b	121
g	c	139
g	d	143
g	e	100
g	f	123
g	g	0
```

```python
assert ["id", "address"] in clusters
assert ["a", "1.1.1.1.1.1.1.1.1.1.1.1.1.1"] in clusters
assert ["b", "1.1.1.1.1.2.2.2.2.2.2.2.2.2"] in clusters
assert ["c", "1.1.1.2.2.3.3.3.3.3.3.3.3.3"] in clusters
assert ["d", "2.2.2.4.4.5.5.5.5.5.5.5.5.5"] in clusters
assert ["e", "1.1.1.3.3.4.4.4.4.4.4.4.4.4"] in clusters
assert ["f", "1.1.1.1.1.1.1.1.1.1.1.1.1.1"] in clusters
assert ["g", "3.3.3.5.5.6.6.6.6.6.6.6.6.6"] in clusters
```

**Review by apetkau — APPROVED** [review]
Thanks so much Eric. This looks great 😄
